File: backend/app/schemas/classifications.py
from typing import Optional, List, ForwardRef
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime

class ClassificationBase(BaseModel):
    code: str = Field(..., max_length=50)
    name: str = Field(..., max_length=150)
    description: Optional[str] = None
    level: int = 0
    status: str = "ACTIVE"

class ClassificationCreate(ClassificationBase):
    parent_id: Optional[str] = None

class ClassificationUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None

class ClassificationMove(BaseModel):
    new_parent_id: Optional[str] = None

class ClassificationRead(ClassificationBase):
    classification_id: str
    parent_id: Optional[str] = None
    version: int
    is_latest: bool

    model_config = ConfigDict(from_attributes=True)

ClassificationNode = ForwardRef('ClassificationNode')

class ClassificationNode(ClassificationRead):
    children: List[ClassificationNode] = Field(default_factory=list)

ClassificationNode.model_rebuild()

File: frontend/src/api/classifications.ts
import { apiClient } from './apiClient';

export interface Classification {
  classification_id: string;
  code: string;
  name: string;
  description: string;
  level: number;
  status: string;
  parent_id: string | null;
  version: number;
  is_latest: boolean;
}

export interface ClassificationNode extends Classification {
  children: ClassificationNode[];
}

export interface ClassificationCreate {
  code: string;
  name: string;
  parent_id?: string | null;
  description?: string | null;
}

export const getClassificationsFn = async (): Promise<Classification[]> => {
  const response = await apiClient.get<Classification[]>('/classifications');
  return response.data;
};

export const getClassificationTreeFn = async (): Promise<ClassificationNode[]> => {
  const response = await apiClient.get<ClassificationNode[]>('/classifications/tree');
  return response.data;
};

export const createClassificationFn = async (payload: ClassificationCreate): Promise<Classification> => {
  const response = await apiClient.post<Classification>('/classifications', payload);
  return response.data;
};

export const deactivateClassificationFn = async (id: string): Promise<Classification> => {
  const response = await apiClient.post<Classification>(`/classifications/${id}/deactivate`);
  return response.data;
};
